### ISSUE TITLE
Max simultaneous connections for user 1000

### ISSUE THREAD

**Issue by tom-seechange** [Azure/iot-identity-service#433]
Hi. When using aziot-edge 1.3.0 with aziot-identity 1.3.0 our modules fail to deploy with the following errors:
```

<6> 2022-07-26 13:58:22.635 +00:00 [INF] [Microsoft.Azure.Devices.Edge.Agent.Core.Planners.HealthRestartPlanner] - Unable to process module x add or update as the module identity could not be obtained
<6> 2022-07-26 13:58:22.635 +00:00 [INF] [Microsoft.Azure.Devices.Edge.Agent.Core.Planners.HealthRestartPlanner] - Unable to process module y add or update as the module identity could not be obtained
<6> 2022-07-26 13:58:22.635 +00:00 [INF] [Microsoft.Azure.Devices.Edge.Agent.Core.Planners.HealthRestartPlanner] - Unable to process module z add or update as the module identity could not be obtained
<6> 2022-07-26 13:58:22.635 +00:00 [INF] [Microsoft.Azure.Devices.Edge.Agent.Core.Planners.HealthRestartPlanner] - Unable to process module a add or update as the module identity could not be obtained
<6> 2022-07-26 13:58:22.635 +00:00 [INF] [Microsoft.Azure.Devices.Edge.Agent.Core.Planners.HealthRestartPlanner] - Unable to process module edgeHub add or update as the module identity could not be obtained
<6> 2022-07-26 13:58:22.635 +00:00 [INF] [Microsoft.Azure.Devices.Edge.Agent.Core.Planners.HealthRestartPlanner] - Unable to process module b add or update as the module identity could not be obtained
<6> 2022-07-26 13:58:22.635 +00:00 [INF] [Microsoft.Azure.Devices.Edge.Agent.Core.Planners.HealthRestartPlanner] - Unable to process module c add or update as the module identity could not be obtained
<6> 2022-07-26 13:58:22.635 +00:00 [INF] [Microsoft.Azure.Devices.Edge.Agent.Core.Planners.HealthRestartPlanner] - Unable to process module d add or update as the module identity could not be obtained
<6> 2022-07-26 13:58:22.635 +00:00 [INF] [Microsoft.Azure.Devices.Edge.Agent.Core.Planners.HealthRestartPlanner] - Unable to process module e add or update as the module identity could not be obtained
<6> 2022-07-26 13:58:22.635 +00:00 [INF] [Microsoft.Azure.Devices.Edge.Agent.Core.Planners.HealthRestartPlanner] - Unable to process module f add or update as the module identity could not be obtained
<6> 2022-07-26 13:58:22.635 +00:00 [INF] [Microsoft.Azure.Devices.Edge.Agent.Core.Planners.HealthRestartPlanner] - Unable to process module g add or update as the module identity could not be obtained
<6> 2022-07-26 13:58:22.635 +00:00 [INF] [Microsoft.Azure.Devices.Edge.Agent.Core.Planners.HealthRestartPlanner] - Unable to process module h add or update as the module identity could not be obtained
```

This is caused by timeouts calling the identity service:

```
7> 2022-07-26 13:58:27.724 +00:00 [DBG] [Microsoft.Azure.Devices.Edge.Agent.Edgelet.ModuleManagementHttpClient] - Error when getting an Http response from unix:///var/run/iotedge/mgmt.sock for Update identity for f with generation ID 12345
System.IO.IOException: Unable to read data from the transport connection: Broken pipe.
 ---> System.Net.Sockets.SocketException (32): Broken pipe
   --- End of inner exception stack trace ---
   at System.IO.BufferedStream.FlushWriteAsync(CancellationToken cancellationToken)
   at System.IO.BufferedStream.ReadFromUnderlyingStreamAsync(Memory`1 buffer, CancellationToken cancellationToken, Int32 bytesAlreadySatisfied, Task semaphoreLockTask)
   at Microsoft.Azure.Devices.Edge.Util.Uds.HttpBufferedStream.ReadLineAsync(CancellationToken cancellationToken) in /home/vsts/work/1/s/edge-util/src/Microsoft.Azure.Devices.Edge.Util/uds/HttpBufferedStream.cs:line 62
   at Microsoft.Azure.Devices.Edge.Util.Uds.HttpRequestResponseSerializer.SetResponseStatusLine(HttpResponseMessage httpResponse, HttpBufferedStream bufferedStream, CancellationToken cancellationToken) in /home/vsts/work/1/s/edge-util/src/Microsoft.Azure.Devices.Edge.Util/uds/HttpRequestResponseSerializer.cs:line 158
   at Microsoft.Azure.Devices.Edge.Util.Uds.HttpRequestResponseSerializer.DeserializeResponse(HttpBufferedStream bufferedStream, CancellationToken cancellationToken) in /home/vsts/work/1/s/edge-util/src/Microsoft.Azure.Devices.Edge.Util/uds/HttpRequestResponseSerializer.cs:line 69
   at Microsoft.Azure.Devices.Edge.Util.Uds.HttpUdsMessageHandler.SendAsync(HttpRequestMessage request, CancellationToken cancellationToken) in /home/vsts/work/1/s/edge-util/src/Microsoft.Azure.Devices.Edge.Util/uds/HttpUdsMessageHandler.cs:line 40
   at System.Net.Http.HttpClient.FinishSendAsyncUnbuffered(Task`1 sendTask, HttpRequestMessage request, CancellationTokenSource cts, Boolean disposeCts)
   at Microsoft.Azure.Devices.Edge.Agent.Edgelet.Version_2020_07_07.GeneratedCode.EdgeletHttpClient.UpdateIdentityAsync(String api_version, String name, UpdateIdentity updateinfo, CancellationToken cancellationToken) in /home/vsts/work/1/s/edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Edgelet/version_2020_07_07/generatedCode/EdgeletHttpClient.cs:line 1142
   at Microsoft.Azure.Devices.Edge.Util.TaskEx.TimeoutAfter[T](Task`1 task, TimeSpan timeout) in /home/vsts/work/1/s/edge-util/src/Microsoft.Azure.Devices.Edge.Util/TaskEx.cs:line 128
   at Microsoft.Azure.Devices.Edge.Agent.Edgelet.Versioning.ModuleManagementHttpClientVersioned.Execute[T](Func`1 func, String operation) in /home/vsts/work/1/s/edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Edgelet/versioning/ModuleManagementHttpClientVersioned.cs:line 146
```

We have around 15 modules deployed and this seems to cause the edgeAgent to create too many simultaneous connections to the identity service since the per uid limit of 10 was put in place:

[https://github.com/Azure/iot-identity-service/blob/release/1.3/http-common/src/connector.rs#L146](https://github.com/Azure/iot-identity-service/blob/release/1.3/http-common/src/connector.rs#L146)

Message from our logs:

`Jul 26 14:03:31 thedevicename aziot-edged[16455]: 2022-07-26T14:03:31Z [INFO] - Max simultaneous connections reached for user 1000`

Downgrading edge to 1.2.9 and identity to 1.2.6 gets it going again following a full restart of all services.

This limit seems to cause our entire deployment to start failing as the timeout isn't handled gracefully. We end up with no modules running except the edgeAgent.

Thanks,

Tom

**Comment by arsing** [comment]
Hmm, I believe we had already thought about this when writing the feature and had a reason that it shouldn't have been a problem. But we'll check again.

**Comment by StephanZaat** [comment]
We have the same problem. This is mostly happening at edge locations where the network connection is a bit slower and the initial module download size is quite high.

**Comment by arsing** [comment]
This was fixed by 3ce096b5b42b953a9fbac6857d78d3a4c6021c68 + d966a2227f3de42e5324db93786c572b203da605

**Comment by HahUK** [comment]
Thanks very much for sorting it arsing and gordonwang0. Do you have a rough ETA for merging and then releasing?

Thanks again,

Tom

**Comment by micahl** [comment]
Closing.  This should be fixed in the recent 1.4 release.
